dsl_parser/framework/parser.py

```diff
@@ -148,32 +162,111 @@ def _traverse_schema(self, schema, parent_element):
         else:
             raise ValueError('Illegal state should have been identified'
                              ' by schema API validation')
+        return parent_element
 
-    def _traverse_dict_schema(self, schema, parent_element):
+    def _traverse_dict_schema(self, schema, parent_element, namespace):
         if not isinstance(parent_element.initial_value, dict):
             return
 
         parsed_names = set()
         for name, element_cls in schema.items():
             if name not in parent_element.initial_value_holder:
                 value = None
+                current_namespace = namespace
             else:
                 name, value = \
                     parent_element.initial_value_holder.get_item(name)
                 parsed_names.add(name.value)
+                current_namespace = value.namespace or namespace
             self._traverse_element_cls(element_cls=element_cls,
                                        name=name,
                                        value=value,
-                                       parent_element=parent_element)
+                                       parent_element=parent_element,
+                                       namespace=current_namespace)
         for k_holder, v_holder in parent_element.initial_value_holder.value.\
                 iteritems():
             if k_holder.value not in parsed_names:
+                current_namespace = v_holder.namespace or namespace
                 self._traverse_element_cls(element_cls=elements.UnknownElement,
-                                           name=k_holder, value=v_holder,
-                                           parent_element=parent_element)
+                                           name=k_holder,
+                                           value=v_holder,
+                                           parent_element=parent_element,
+                                           namespace=current_namespace)
+
+    def _traverse_element_type_schema(self, schema, parent_element, namespace):
```

Comment:
I've left a bunch of specific notes below, but this function requires serious refactoring in general, so I'm leaving it up to you to heed my advices or choose your own way.